accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -885,10 +885,7 @@ one of the following criteria holds:
 
 - `T` is an extension type that does not implement `Future`.
 - `T` is `S?`, and `S` is incompatible with await.
-- `T` is `X & B`, and either:
-  - `B` is incompatible with await, or
-  - `B` does not derive a future type, and `X` is
-    incompatible with await.
+- `T` is `X & B`, and `B` is incompatible with await.
```

Comment:
Should be sufficient.

If `B` is compatible with await, then either `B`:
* is not a (nullable-bounded-)extension type, which means the bound of `X` is also not an extension type. In which case *flatten* will not yield a (nullable-)extension type.
* or is an extension type which implements `Future`, in which case that's the future which will be awaited as expected.

In either case, the *flatten* doing "if `B` does not derive a future type, use *flatten*(`X`)` will not end up using a non-future-extension-type as a type to flatten, which is precisely what we want: The user never needing to know how to flatten a non-future-extension-type.

So LGTM!